openwisp_monitoring/device/admin.py

```diff
@@ -48,26 +48,70 @@ def full_clean(self):
             obj = form.instance
             if not obj.content_type or not obj.object_id:
                 setattr(
-                    form.instance,
+                    obj,
                     self.ct_field.get_attname(),
                     ContentType.objects.get_for_model(self.instance).pk,
                 )
-                setattr(form.instance, self.ct_fk_field.get_attname(), self.instance.pk)
+                setattr(obj, self.ct_fk_field.get_attname(), self.instance.pk)
         super().full_clean()
 
+    def save_new(self, form, commit=True):
+        instance = super().save_new(form, commit=False)
```

Comment:
is it possible to obtain the same by moving this logic to the model instead of here?
The problem with this code is that if we implement similar logic in the REST API or in other views, we will have to repeat it.

Reply:
No problem, I'll move it to their respective models 👍